`opam switch create` with `static` option fails on Ubuntu aarch64

Consider this Dockerfile:

```docker
FROM ubuntu:latest

RUN apt update
RUN apt install -y --no-install-recommends ca-certificates git rsync opam musl-tools
RUN opam init -y --bare --disable-sandboxing
RUN opam switch create 5.2.0 --packages ocaml-option-static
```

On x64, it works perfectly fine.

However, on aarch64, `opam switch create` gives me the following error:

```
11.15 [ERROR] The compilation of ocaml-variants.5.2.0+options failed at "make -j9".
11.16
11.16 #=== ERROR while compiling ocaml-variants.5.2.0+options =======================#
11.16 # context              2.1.5 | linux/arm64 | ocaml-option-static.1 | https://opam.ocaml.org#f11abb65
11.16 # path                 ~/.opam/5.2.0/.opam-switch/build/ocaml-variants.5.2.0+options
11.16 # command              /usr/bin/make -j9
11.16 # exit-code            2
11.16 # env-file             ~/.opam/log/ocaml-variants-7-75a8ef.env
11.16 # output-file          ~/.opam/log/ocaml-variants-7-75a8ef.out
11.16 ### output ###
11.16 # make[1]: Entering directory '/root/.opam/5.2.0/.opam-switch/build/ocaml-variants.5.2.0+options'
11.16 # [...]
11.16 #   CC runtime/interp.b.o
11.16 #   CC runtime/startup_byt.b.o
11.16 #   CC runtime/zstd.b.o
11.16 #   MKLIB runtime/libcamlrun.a
11.16 #   MKEXE runtime/ocamlrun
11.16 # /usr/bin/ld: /usr/lib/gcc/aarch64-linux-gnu/13/libgcc.a(lse-init.o): in function `init_have_lse_atomics':
11.16 # (.text.startup+0xc): undefined reference to `__getauxval'
11.16 # collect2: error: ld returned 1 exit status
11.16 # make[1]: *** [Makefile:1309: runtime/ocamlrun] Error 1
11.16 # make[1]: Leaving directory '/root/.opam/5.2.0/.opam-switch/build/ocaml-variants.5.2.0+options'
11.16 # make: *** [Makefile:830: world.opt] Error 2
```

--- a/packages/ocaml-variants/ocaml-variants.5.0.0+options/opam
+++ b/packages/ocaml-variants/ocaml-variants.5.0.0+options/opam
@@ -85,7 +85,8 @@ build: [
     "--enable-frame-pointers" {ocaml-option-fp:installed}
     "CC=cc" {!ocaml-option-32bit:installed & !ocaml-option-musl:installed & (os="openbsd"|os="macos")}
     "CC=musl-gcc" {ocaml-option-musl:installed & os-distribution!="alpine"}
-    "CFLAGS=-Os" {ocaml-option-musl:installed}
+    "CFLAGS=-Os" {ocaml-option-musl:installed & arch!="arm64"}
+    "CFLAGS=-Os -mno-outline-atomics" {ocaml-option-musl:installed & arch="arm64"}
     "LDFLAGS=-Wl,--no-as-needed,-ldl" {ocaml-option-leak-sanitizer:installed | (ocaml-option-address-sanitizer:installed & os!="macos")}
     "CC=gcc -ldl -fsanitize=leak -fno-omit-frame-pointer -O1 -g" {ocaml-option-leak-sanitizer:installed}
     "CC=gcc -ldl -fsanitize=address -fno-omit-frame-pointer -O1 -g" {ocaml-option-address-sanitizer:installed & os!="macos"}

--- a/packages/ocaml-variants/ocaml-variants.5.1.1+options/opam
+++ b/packages/ocaml-variants/ocaml-variants.5.1.1+options/opam
@@ -86,7 +86,8 @@ build: [
     "--without-zstd" {ocaml-option-no-compression:installed}
     "CC=cc" {!ocaml-option-32bit:installed & !ocaml-option-musl:installed & (os="openbsd"|os="macos")}
     "CC=musl-gcc" {ocaml-option-musl:installed & os-distribution!="alpine"}
-    "CFLAGS=-Os" {ocaml-option-musl:installed}
+    "CFLAGS=-Os" {ocaml-option-musl:installed & arch!="arm64"}
+    "CFLAGS=-Os -mno-outline-atomics" {ocaml-option-musl:installed & arch="arm64"}
     "LDFLAGS=-Wl,--no-as-needed,-ldl" {ocaml-option-leak-sanitizer:installed | (ocaml-option-address-sanitizer:installed & os!="macos")}
     "CC=gcc -ldl -fsanitize=leak -fno-omit-frame-pointer -O1 -g" {ocaml-option-leak-sanitizer:installed}
     "CC=gcc -ldl -fsanitize=address -fno-omit-frame-pointer -O1 -g" {ocaml-option-address-sanitizer:installed & os!="macos"}

--- a/packages/ocaml-variants/ocaml-variants.5.2.0+options/opam
+++ b/packages/ocaml-variants/ocaml-variants.5.2.0+options/opam
@@ -88,7 +88,8 @@ build: [
     "CC=cc" {!ocaml-option-32bit:installed & !ocaml-option-musl:installed & (os="openbsd"|os="macos")}
     "CC=clang" {ocaml-option-tsan:installed & (os="macos")}
     "CC=musl-gcc" {ocaml-option-musl:installed & os-distribution!="alpine"}
-    "CFLAGS=-Os" {ocaml-option-musl:installed}
+    "CFLAGS=-Os" {ocaml-option-musl:installed & arch!="arm64"}
+    "CFLAGS=-Os -mno-outline-atomics" {ocaml-option-musl:installed & arch="arm64"}
     "LDFLAGS=-Wl,--no-as-needed,-ldl" {ocaml-option-leak-sanitizer:installed | (ocaml-option-address-sanitizer:installed & os!="macos")}
     "CC=gcc -ldl -fsanitize=leak -fno-omit-frame-pointer -O1 -g" {ocaml-option-leak-sanitizer:installed}
     "CC=gcc -ldl -fsanitize=address -fno-omit-frame-pointer -O1 -g" {ocaml-option-address-sanitizer:installed & os!="macos"}

